### DISCUSSION TITLE
REQ: optional parameter for image metadata

### DISCUSSION THREAD

**Discussion by bblackmoor** [lllyasviel/Fooocus#2132] [Ideas]
"Fooocus is committed to protect user privacy and will never write any hidden data in your generated images."
-- https://github.com/lllyasviel/Fooocus/discussions/160

Allowing a user to **choose** to include the _image_ metadata (prompt, model, seed, etc.) of the images they generate is no more an invasion of privacy than is the image itself. (It may be helpful to refer to GDPR for a useful definition of "personal data".)
-- https://gdpr-info.eu/issues/personal-data/

Would you please include an _optional_ parameter which allows the user to **choose** to include the metadata of the images they generate?

For example, something like this in config.text?

`"include_image_metadata": 1,`

Thank you for your time.

**Comment by bblackmoor** [comment]
Here is an alternative setting, if you would like to make it crystal clear to users that _image_ metadata is not _personal_ data:

`"include_image_metadata_but_no_gpdr_personal_information": 1,`

**Comment by mashb1t** [comment]
Yesterday i've finished the implementation of https://github.com/lllyasviel/Fooocus/pull/1940, please check it out. Still marked as WIP as this is in final internal QA now.

**Reply by mashb1t** [reply]
it's ready now. If you already want to use t you can check out my fork at https://github.com/mashb1t/Fooocus

**Reply by bblackmoor** [reply]
Do you have any idea when it will be merged into Fooocus?

**Reply by mashb1t** [reply]
No, sadly not. I'm not a fan of merging my own changes without proper review, so i hope that somebody of lllyasviels team or he himself is going to have a look soon.

**Reply by mashb1t** [reply]
Still no fan, but after community feedback and extensive testing the feature is now in the official releases.

**Comment by bblackmoor** [comment]
Closed as a duplicate of https://github.com/lllyasviel/Fooocus/discussions/160
